opal/Makefile.am

```diff
@@ -91,6 +91,14 @@ lib@OPAL_LIB_NAME@_la_LDFLAGS = -version-info @libopen_pal_so_version@ \
 	$(opal_hwloc_LDFLAGS) \
 	$(opal_pmix_LDFLAGS)
 
+if OPAL_cuda_support
+if MCA_BUILD_opal_DSO
+lib@OPAL_LIB_NAME@_la_LIBADD += \
+	mca/common/cuda/lib@OPAL_LIB_NAME@mca_common_cuda.la
```

Comment:
This isn't right.  If the code is in mca/common, it's a common library and can't be special cased like this.  If it needs to be in libopen-pal.{a,so} all the time (as this patch suggests), then it doesn't belong in a common component and belongs in the base library.

Reply:
Hmm, I can revert the moving of code from datatype -> common then if that's the case

Reply:
William and I talked about this offline, but some notes for posterity.  There are some corner cases around common code in the core library that aren't worth fixing, so we talked about two possibilities:

1) If the problem is that it's not datatype code, then move it to another convenience library outside of common.  Like util or cuda or gpu or something at the top level of opal.

2) If the problem is you want to create an optional interface that is used by the core opal/ompi libraries, then you have to make a new framework, not use common.  The "must always be present" code then goes in the framework base, and the